[Bug]: Two tests fail with Python 3.14 due to unexpected reference count

### What Operating System(s) are you seeing this problem on?

Linux (x86-64)

### dlib version

20.0

### Python version

3.14.0rc1

### Compiler

GCC 15

### Expected Behavior

All test should pass as they do when run with Python 3.13.5.

### Current Behavior

Building dlib with Python 3.14.0rc1 for Fedora we noticed two tests failing:

```python
=================================== FAILURES ===================================
___________________ test_regression_issue_1220_get_face_chip ___________________
    @pytest.mark.skipif(not utils.is_numpy_installed(), reason="requires numpy")
    def test_regression_issue_1220_get_face_chip():
        """
        Memory leak in Python get_face_chip
        https://github.com/davisking/dlib/issues/1220
        """
        face_chip = get_test_face_chip()
        # we expect two references:
        # 1.) the local variable
        # 2.) the temporary passed to getrefcount
>       assert sys.getrefcount(face_chip) == 2
E       assert 1 == 2
E        +  where 1 = <built-in function getrefcount>(array([[[ 83,  57,  31],\n        [ 83,  58,  31],\n        [ 83,  57,  32],\n        ...,\n        [ 42,  15,   6],\n        [118,  84,  55],\n        [  0,   0,   0]],\n\n       [[ 75,  54,  32],\n        [ 76,  54,  30],\n        [ 77,  54,  29],\n        ...,\n        [ 87,  57,  35],\n        [135,  97,  64],\n        [  0,   0,   0]],\n\n       [[ 60,  44,  28],\n        [ 60,  45,  28],\n        [ 64,  46,  27],\n        ...,\n        [103,  72,  41],\n        [142, 104,  64],\n        [  0,   0,   0]],\n\n       ...,\n\n       [[  0,   0,   0],\n        [  0,   0,   0],\n        [  0,   0,   0],\n        ...,\n        [ 30,  32,  44],\n        [ 33,  35,  47],\n        [ 35,  37,  49]],\n\n       [[  0,   0,   0],\n        [  0,   0,   0],\n        [  0,   0,   0],\n        ...,\n        [ 26,  28,  40],\n        [ 31,  33,  45],\n        [ 34,  36,  48]],\n\n       [[  0,   0,   0],\n        [  0,   0,   0],\n        [  0,   0,   0],\n        ...,\n        [ 23,  25,  37],\n        [ 27,  29,  41],\n        [ 30,  32,  44]]], shape=(150, 150, 3), dtype=uint8))
E        +    where <built-in function getrefcount> = sys.getrefcount
tools/python/test/test_numpy_returns.py:59: AssertionError
__________________ test_regression_issue_1220_get_face_chips ___________________
    @pytest.mark.skipif(not utils.is_numpy_installed(), reason="requires numpy")
    def test_regression_issue_1220_get_face_chips():
        """
        Memory leak in Python get_face_chip
        https://github.com/davisking/dlib/issues/1220
        """
        face_chips = get_test_face_chips()
        count = sys.getrefcount(face_chips)
>       assert count == 2
E       assert 1 == 2
tools/python/test/test_numpy_returns.py:70: AssertionError
=========================== short test summary info ============================
FAILED tools/python/test/test_numpy_returns.py::test_regression_issue_1220_get_face_chip - assert 1 == 2
 +  where 1 = <built-in function getrefcount>(array([[[ 83,  57,  31],\n        [ 83,  58,  31],\n        [ 83,  57,  32],\n        ...,\n        [ 42,  15,   6],\n        [118,  84,  55],\n        [  0,   0,   0]],\n\n       [[ 75,  54,  32],\n        [ 76,  54,  30],\n        [ 77,  54,  29],\n        ...,\n        [ 87,  57,  35],\n        [135,  97,  64],\n        [  0,   0,   0]],\n\n       [[ 60,  44,  28],\n        [ 60,  45,  28],\n        [ 64,  46,  27],\n        ...,\n        [103,  72,  41],\n        [142, 104,  64],\n        [  0,   0,   0]],\n\n       ...,\n\n       [[  0,   0,   0],\n        [  0,   0,   0],\n        [  0,   0,   0],\n        ...,\n        [ 30,  32,  44],\n        [ 33,  35,  47],\n        [ 35,  37,  49]],\n\n       [[  0,   0,   0],\n        [  0,   0,   0],\n        [  0,   0,   0],\n        ...,\n        [ 26,  28,  40],\n        [ 31,  33,  45],\n        [ 34,  36,  48]],\n\n       [[  0,   0,   0],\n        [  0,   0,   0],\n        [  0,   0,   0],\n        ...,\n        [ 23,  25,  37],\n        [ 27,  29,  41],\n        [ 30,  32,  44]]], shape=(150, 150, 3), dtype=uint8))
 +    where <built-in function getrefcount> = sys.getrefcount
FAILED tools/python/test/test_numpy_returns.py::test_regression_issue_1220_get_face_chips - assert 1 == 2
========================= 2 failed, 73 passed in 0.69s =========================
```

### Steps to Reproduce

`uc venv --python 3.14 _py314`
`source _py314/bin activate`
`uv pip install -e .`
`uv pip install pytest numpy`
`pytest -r fEs`

### Anything else?

There is some mention of changed reference counts in the [Python 3.14 release notes](https://docs.python.org/3.14/whatsnew/3.14.html#id13).

--- a/tools/python/test/test_numpy_returns.py
+++ b/tools/python/test/test_numpy_returns.py
@@ -56,7 +56,14 @@ def test_regression_issue_1220_get_face_chip():
     # we expect two references:
     # 1.) the local variable
     # 2.) the temporary passed to getrefcount
-    assert sys.getrefcount(face_chip) == 2
+    #
+    # Python 3.14 has changed the way references are counted (borrowed references)
+    # https://docs.python.org/3.14/whatsnew/3.14.html#limited-c-api-changes
+    # https://github.com/davisking/dlib/issues/3096
+    if sys.version_info < (3, 14):
+        assert sys.getrefcount(face_chip) == 2
+    else:
+        assert sys.getrefcount(face_chip) == 1
 
 
 @pytest.mark.skipif(not utils.is_numpy_installed(), reason="requires numpy")
@@ -67,6 +74,12 @@ def test_regression_issue_1220_get_face_chips():
     """
     face_chips = get_test_face_chips()
     count = sys.getrefcount(face_chips)
-    assert count == 2
+    # Python 3.14 has changed the way references are counted (borrowed references)
+    # https://docs.python.org/3.14/whatsnew/3.14.html#limited-c-api-changes
+    # https://github.com/davisking/dlib/issues/3096
+    if sys.version_info < (3, 14):
+        assert count == 2
+    else:
+        assert count == 1
     count = sys.getrefcount(face_chips[0])
     assert count == 2

